diff_drive_controller/src/diff_drive_controller.cpp

```diff
@@ -1085,10 +1096,13 @@ namespace diff_drive_controller
   {
     if (isRunning())
     {
+      prev_command_struct_ = command_struct_;
```

Comment:
Why do we have this realtime buffer if we write it into a class attribute anyway? Or is this a situation of one thread owns `prev_command_struct_` and `command_struct_` and the other thread only uses `update` and the communication between the two threads goes through the `command_` realtime buffer?

Just wondering, obviously no need to change this as it's been working for years.

Reply:
The idea is that the consumer that does the read is a real time thread, and the producer the callback is the non real time thread. With this buffer the real time thread will have priority while the non real time thread will wait. In theory this allows the real time thread to maintain consistent timings. In practice it makes very little difference for our use cases.